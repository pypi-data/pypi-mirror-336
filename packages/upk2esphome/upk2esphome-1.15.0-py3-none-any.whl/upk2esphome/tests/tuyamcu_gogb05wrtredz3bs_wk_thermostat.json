{
	"category": "wk",
	"model": {
		"modelId": "00000045od",
		"services": [
			{
				"actions": [],
				"code": "",
				"description": "",
				"events": [],
				"name": "默认服务",
				"properties": [
					{
						"abilityId": 1,
						"accessMode": "rw",
						"code": "switch",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_power3"
						},
						"name": "开关",
						"typeSpec": {
							"type": "bool",
							"typeDefaultValue": false
						}
					},
					{
						"abilityId": 2,
						"accessMode": "rw",
						"code": "mode",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_mode"
						},
						"name": "工作模式",
						"typeSpec": {
							"range": [
								"auto",
								"manual"
							],
							"type": "enum",
							"typeDefaultValue": "auto"
						}
					},
					{
						"abilityId": 10,
						"accessMode": "rw",
						"code": "frost",
						"description": "",
						"extensions": {
							"iconName": "icon-chushuang"
						},
						"name": "防霜冻功能",
						"typeSpec": {
							"type": "bool",
							"typeDefaultValue": false
						}
					},
					{
						"abilityId": 16,
						"accessMode": "rw",
						"code": "temp_set",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_temp"
						},
						"name": "目标温度",
						"typeSpec": {
							"max": 90,
							"min": 5,
							"scale": 0,
							"step": 1,
							"type": "value",
							"typeDefaultValue": 5,
							"unit": "℃"
						}
					},
					{
						"abilityId": 19,
						"accessMode": "rw",
						"code": "upper_temp",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_f"
						},
						"name": "设置温度上限",
						"typeSpec": {
							"max": 90,
							"min": 30,
							"scale": 0,
							"step": 1,
							"type": "value",
							"typeDefaultValue": 30,
							"unit": "℃"
						}
					},
					{
						"abilityId": 24,
						"accessMode": "ro",
						"code": "temp_current",
						"description": "",
						"extensions": {
							"iconName": "icon-wendu"
						},
						"name": "当前温度",
						"typeSpec": {
							"max": 900,
							"min": 0,
							"scale": 1,
							"step": 5,
							"type": "value",
							"typeDefaultValue": 0,
							"unit": "℃"
						}
					},
					{
						"abilityId": 26,
						"accessMode": "rw",
						"code": "lower_temp",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_c"
						},
						"name": "设置温度下限",
						"typeSpec": {
							"max": 20,
							"min": 5,
							"scale": 0,
							"step": 1,
							"type": "value",
							"typeDefaultValue": 5,
							"unit": "℃"
						}
					},
					{
						"abilityId": 27,
						"accessMode": "rw",
						"code": "temp_correction",
						"description": "",
						"extensions": {
							"iconName": "icon-wendu1"
						},
						"name": "温度校准",
						"typeSpec": {
							"max": 9,
							"min": -9,
							"scale": 0,
							"step": 1,
							"type": "value",
							"typeDefaultValue": -9,
							"unit": "摄氏度"
						}
					},
					{
						"abilityId": 36,
						"accessMode": "ro",
						"code": "valve_state",
						"description": "",
						"extensions": {
							"iconName": "icon-tcl_function_vs2"
						},
						"name": "阀门状态",
						"typeSpec": {
							"range": [
								"open",
								"close"
							],
							"type": "enum",
							"typeDefaultValue": "open"
						}
					},
					{
						"abilityId": 38,
						"accessMode": "rw",
						"code": "week_program1",
						"description": "",
						"extensions": {
							"iconName": "icon-icon-test7"
						},
						"name": "周编程（自由式）",
						"typeSpec": {
							"maxlen": 128,
							"type": "raw"
						}
					},
					{
						"abilityId": 39,
						"accessMode": "rw",
						"code": "factory_reset",
						"description": "",
						"extensions": {
							"iconName": "icon-icon-test12"
						},
						"name": "恢复出厂设置",
						"typeSpec": {
							"type": "bool",
							"typeDefaultValue": false
						}
					},
					{
						"abilityId": 40,
						"accessMode": "rw",
						"code": "child_lock",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_lock"
						},
						"name": "童锁开关",
						"typeSpec": {
							"type": "bool",
							"typeDefaultValue": false
						}
					},
					{
						"abilityId": 43,
						"accessMode": "rw",
						"code": "sensor_choose",
						"description": "",
						"extensions": {
							"iconName": "icon-gongneng"
						},
						"name": "传感器选择",
						"typeSpec": {
							"range": [
								"in",
								"out",
								"all"
							],
							"type": "enum",
							"typeDefaultValue": "in"
						}
					},
					{
						"abilityId": 45,
						"accessMode": "ro",
						"code": "fault",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_notice"
						},
						"name": "故障告警",
						"typeSpec": {
							"label": [
								"e1",
								"e2",
								"e3"
							],
							"maxlen": 3,
							"type": "bitmap",
							"typeDefaultValue": 0
						}
					},
					{
						"abilityId": 101,
						"accessMode": "rw",
						"code": "Temp_dif",
						"description": "",
						"name": "温差启动",
						"typeSpec": {
							"max": 5,
							"min": 1,
							"scale": 0,
							"step": 1,
							"type": "value",
							"typeDefaultValue": 1,
							"unit": "℃"
						}
					},
					{
						"abilityId": 102,
						"accessMode": "rw",
						"code": "program_mode",
						"description": "",
						"name": "编程模式",
						"typeSpec": {
							"range": [
								"00",
								"52",
								"61",
								"70"
							],
							"type": "enum",
							"typeDefaultValue": "00"
						}
					}
				]
			}
		]
	}
}
