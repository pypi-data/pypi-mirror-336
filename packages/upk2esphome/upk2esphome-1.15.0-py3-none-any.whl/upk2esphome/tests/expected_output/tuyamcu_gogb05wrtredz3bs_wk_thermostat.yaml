uart:
  rx_pin: RX1
  tx_pin: TX1
  baud_rate: 115200

tuya:
  # DPIDs processed from schema model: 00000045od
  on_datapoint_update:
    - sensor_datapoint: 36
      datapoint_type: enum
      then:
        - text_sensor.template.publish:
            id: tuya_valve_state
            state: !lambda "return std::to_string(x);"

switch:
  - platform: tuya
    switch_datapoint: 1
    name: (Unconfirmed) Switch
  - platform: tuya
    switch_datapoint: 10
    name: (Unconfirmed) Frost
  - platform: tuya
    switch_datapoint: 39
    name: (Unconfirmed) Factory Reset
  - platform: tuya
    switch_datapoint: 40
    name: (Unconfirmed) Child Lock

select:
  - platform: tuya
    enum_datapoint: 2
    name: (Unconfirmed) Mode
    optimistic: true
    options:
      0: Auto
      1: Manual
  - platform: tuya
    enum_datapoint: 43
    name: (Unconfirmed) Sensor Choose
    optimistic: true
    options:
      0: In
      1: Out
      2: All
  - platform: tuya
    enum_datapoint: 102
    name: (Unconfirmed) Program Mode
    optimistic: true
    options:
      0: "00"
      1: "52"
      2: "61"
      3: "70"

number:
  - platform: tuya
    number_datapoint: 16
    name: (Unconfirmed) Temp Set
    unit_of_measurement: "\xB0C"
    min_value: 5
    max_value: 90
    step: 1
  - platform: tuya
    number_datapoint: 19
    name: (Unconfirmed) Upper Temp
    unit_of_measurement: "\xB0C"
    min_value: 30
    max_value: 90
    step: 1
  - platform: tuya
    number_datapoint: 26
    name: (Unconfirmed) Lower Temp
    unit_of_measurement: "\xB0C"
    min_value: 5
    max_value: 20
    step: 1
  - platform: tuya
    number_datapoint: 27
    name: (Unconfirmed) Temp Correction
    unit_of_measurement: ""
    min_value: -9
    max_value: 9
    step: 1
  - platform: tuya
    number_datapoint: 101
    name: (Unconfirmed) Temp Dif
    unit_of_measurement: "\xB0C"
    min_value: 1
    max_value: 5
    step: 1

sensor:
  - platform: tuya
    sensor_datapoint: 24
    name: Temperature
    device_class: temperature
    unit_of_measurement: "\xB0C"
    accuracy_decimals: 1
    filters:
      - multiply: 0.1

text_sensor:
  - platform: template
    id: tuya_valve_state
    name: (Unconfirmed) Valve State
    filters:
      - map:
          - 0 -> Open
          - 1 -> Close
